### DISCUSSION TITLE
can't find the same video

### DISCUSSION THREAD

**Discussion by OverlandJMC** [aclap-dev/video-downloadhelper#1167] [Help and Q&A]
### Your question

A webpage has more than 10 videos list. However, the app shows only 6 videos. 
The 1st one on the list of the app is not the same one as the 1st one on the webpage list.  Even by looking at the video time of the 1st one on the webpage list., I couldn't find the same-length video on the list of the app.

### Addon & CoApp version and details

version: 9.1.0.43
target: google
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36
{
    "used_history_button": true,
    "download_directory": "C:\\Users\\jcha\\Downloads",
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": true,
    "open_count_store": 5,
    "successfull_dl": 2,
    "dable_attempts": {
        "__serializer_tag": "map",
        "__serializer_value": [
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "www.cpaacademy.org"
                    },
                    {
                        "__serializer_tag": "array",
                        "__serializer_value": [
                            {
                                "__serializer_tag": "primitive",
                                "__serializer_value": 5
                            },
                            {
                                "__serializer_tag": "primitive",
                                "__serializer_value": 1
                            }
                        ]
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "taxmentor.com"
                    },
                    {
                        "__serializer_tag": "array",
                        "__serializer_value": [
                            {
                                "__serializer_tag": "primitive",
                                "__serializer_value": 1
                            },
                            {
                                "__serializer_tag": "primitive",
                                "__serializer_value": 1
                            }
                        ]
                    }
                ]
            }
        ]
    },
    "dable_attempts_last": 1724032015335
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I am just another user.

This might or might not solve your problem.  First, you need to check the VDH setting I'm highlighting in this image:

![#01](https://github.com/user-attachments/assets/9f19c2bd-cc82-401e-b54c-0638d60c857d)

Then, you need to pop up the context menu on one of your videos on your web page.

Then you need to use VDH through that context menu.

That should do the trick.  But if not, you should post the URL of the page you are talking about & I can try to show you via screenshots what to do.

Why is this discussion closed?  Did you figure out how to solve your problem?

**Reply by OverlandJMC** [reply]
Thank you for your tip.  I closed it because it worked when I accessed the page again.

**Reply by GrampaWildWilly** [reply]
Don't you just love it when things work different all the time?

I'm glad to hear things are all good now.

**Comment by OverlandJMC** [comment]
The problem is happening again. Please see the attached screenshot. 
![can't find the same video](https://github.com/user-attachments/assets/dd7d083b-dbfc-45b6-8088-e6d7c7e27acd)
Access commands from right-click in page is checked.
The URL of the page - https://taxrepllc.com/ma-qa-calls/#1706138149134-ac4b615f-767e It is for the members.

**Reply by GrampaWildWilly** [reply]
Too bad it's a subscription-only site.  That makes it difficult for me to properly investigate this.

Do I understand this correctly?  Your image is showing VDH recognizing 8 videos but there are more than 8 videos on the page, is that right?  At least VDH is showing you durations.  Despite it giving you the same name for every one of them, you should be able to match the entries on the VDH menu with individual videos on the page through their unique durations, no?  If the order of the entries on the VDH menu does not match the order of presentation of the videos on the page, that is annoying.  I don't know how VDH decides to populate its menu from this page.  The video on the left side of your image shows a duration of 34:06.  The fifth entry on your VDH menu shows the same duration.  They must match up.  This is the only way I can see for you to figure things out here.

I would guess that if you figure out which videos are not showing on the VDH menu, you could get them to appear by clicking the play button on each of those & then pausing it after a couple of seconds.  I hope.

The reason for checking that box in the settings is so you can use a different way of getting a video.  Instead of looking directly on the VDH menu, pop up the context menu on an individual video.  Click mouse button 2 on the icon for a video to pop up its context menu.  Because of that VDH setting, the context menu will show options for getting that video with VDH.  I would show you a picture but I can't get into the site.  I suppose I could use Pornhub as an example but that's a little too NSFW.  Come to think of it, popping up the context menu might not work.  Some video players respond to a mouse button 2 click by popping up their own menu instead of a standard browser menu.  Again, I can't check this for you but I'm sure you can determine this quickly enough.

YouTube presents a rather strange case.  When you first click mouse button 2 on the video, you get this context menu:

![#01](https://github.com/user-attachments/assets/b218dfe1-dbcd-43a2-a872-1065317d21d9)

But if you just click mouse button 2 again at the exact same place, you get this context menu:

![#02](https://github.com/user-attachments/assets/e93627aa-8719-4e58-9c65-009309d461a9)

That looks like the VDH part of the context menu continues off screen to the bottom.  It actually doesn't.  If I click the play button & pause it after just a second . . .

![#03](https://github.com/user-attachments/assets/205eaf18-3a35-4213-ba06-ee78b807f2a9)

That's weird.  The VDH part of the context menu is gone.  That must be something weird YouTube is doing.  I would hope your web site isn't so anomalous.  YouTube is not necessarily the best place to get examples, but it's certainly the most convenient.

The point I wanted to make is this.  Some video players supply their own context menus.  In such a case, popping up the context menu on a video might not give you useful results.  The example of YouTube shows that you might want to try popping up the context menu more than once, just to see if it changes.  I would hope that on your site, the VDH submenu of the context menu would include useful information you could use to download the video.  Clearly, this doesn't work on YouTube, but YouTube is its own world anyway.  See if this works properly on your site.

**Reply by GrampaWildWilly** [reply]
Just as an aside, I tried this on Pornhub.  Its video player has a captive context menu that does not alternate with a browser menu with repeated clicks of mouse button 2.

Seems to me I saw something that fixes this.  I think it's a Firefox extension that stops video players from doing this.  It gives you a Firefox context menu no matter what.  I think.  My memory is vague.  You could always do a web search on the subject if you're really interested.  But you're using Chrome so you're probably not interested.  Or maybe there is something similar for Chrome.  It's up to you if you want to check this out.

**Reply by GrampaWildWilly** [reply]
Here's an example that is a little better at showing what is possible.  The URL in question is this:

https://www.medici.tv/en/concerts/yannick-nezet-seguin-conducts-berlioz-symphonie-fantastique-vienna-philharmonic-salzburg-festival-2024

It's a subscription site but you don't need to log in to see this page.  It's just a teaser for the full concert.  When you first visit the page, it autoplays a little 5 second ad, then stops on the 36-second teaser showing in this image.  The context menu it gives is this:

![#01](https://github.com/user-attachments/assets/57c92fce-5a34-4cec-a430-5d9b0b5c7028)

But if you click mouse button 2 on the video in a different location, you get this:

![#02](https://github.com/user-attachments/assets/292ead93-2415-463b-9695-7402b0552183)

Notice that the VDH submenu of the context menu offers you variants you can download.  It offers some for the 5-second ad that autoplayed before I took these images.  It also offers some variants for the 36-second teaser.  I would hope your web site offers you something more like this.

**Reply by GrampaWildWilly** [reply]
By the way, you can create a free account on Medici & get to watch & even download the full concert with VDH.
